zb-node.js

```diff
@@ -884,52 +890,87 @@ class ZigbeeNode extends Device {
                          `cmd:${frame.zcl.cmdId}`);
     const profileId = parseInt(frame.profileId, 16);
     const clusterId = parseInt(frame.clusterId, 16);
+    let propertyFound = false;
     for (const property of this.properties.values()) {
       if (profileId == property.profileId &&
           endpoint == property.endpoint &&
           clusterId == property.clusterId) {
+        propertyFound = true;
         switch (frame.zcl.cmdId) {
           case 'on':   // on property
             this.handleButtonOnOffCommand(property, true);
-            this.notifyEvent('1-pressed');
+            this.notifyEvent(`${property.buttonIndex}-pressed`);
+            break;
+          case 'onWithTimedOff':   // on property
+            this.handleButtonOnOffCommand(property, true);
+            this.notifyEvent('motion');
+            this.occupancyTimeout = frame.zcl.payload.ontime / 10;
+            if (this.occupancyTimer) {
+              // remove any previously created timer
+              clearTimeout(this.occupancyTimer);
+            }
+            // create a new timer
+            console.log('Creating a timer for', this.occupancyTimeout, 'seconds');
+            this.occupancyTimer = setTimeout(() => {
+              this.occupancyTimer = null;
+              property.setCachedValue(false);
+              console.log(this.name,
+                          'property:', property.name,
+                          'timeout - clearing value');
+              this.notifyPropertyChanged(property);
+              this.notifyEvent('no-motion');
+            }, this.occupancyTimeout * 1000);
             break;
           case 'off': // on property
           case 'offWithEffect': // onProperty
             this.handleButtonOnOffCommand(property, false);
             this.notifyEvent('2-pressed');
```

Comment:
General question... will the `1-pressed` and `2-pressed` events here always be correct?

Reply:
They are for all of the buttons we currently support. But I modified it to use property.buttonIndex so that it's more consistent with the rest of the code.